src/components/LinterMessage/index.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { Alert } from 'react-bootstrap';
+
+import styles from './styles.module.scss';
+import { LinterMessage as LinterMessageType } from '../../linter';
+
+export type PublicProps = {
+  description: LinterMessageType['description'];
+  message: LinterMessageType['message'];
+  type: LinterMessageType['type'];
+};
+
+const getAlertVariant = (type: PublicProps['type']) => {
+  switch (type) {
+    case 'error':
+      return 'danger';
+    case 'warning':
+      return 'warning';
+    default:
+    case 'notice':
+      return 'secondary';
+  }
+};
+
+const urlPattern = /^https?:\/\//;
+
+const renderDescriptionPart = (part: string) => {
+  // Split a description into words to intercept and replace
+  // URLs with JSX links.
+  return part
+    .trim()
+    .split(' ')
+    .reduce((all: React.ReactNode[], word, index) => {
+      if (index > 0) {
+        // Put back the space.
+        all.push(' ');
+      }
+
+      if (urlPattern.test(word)) {
+        all.push(<Alert.Link href={word}>{word}</Alert.Link>);
+      } else {
+        all.push(word);
+      }
+
+      return all;
+    }, []);
+};
+
+const LinterMessage = ({ description, message, type }: PublicProps) => {
+  return (
+    <Alert variant={getAlertVariant(type)}>
+      <Alert.Heading>{message}</Alert.Heading>
+      <p className={styles.description}>
+        {description.reduce((lines: React.ReactNode[], text, index) => {
+          if (index > 0) {
+            lines.push(<br key={text} />);
+          }
+          lines.push(renderDescriptionPart(text));
+          return lines;
+        }, [])}
```

Comment:
Why is `renderDescriptionPart()` outside the component but not this `reduce()`? It'd be easier to read if we could have a `renderDescription()` function (that could maybe be a single function, embedding the current `renderDescriptionPart()` impl).

Reply:
I moved the reducers into one function and that does read better. Thanks.